rust/datafusion/tests/user_defined_plan.rs

```diff
@@ -178,7 +178,9 @@ async fn topk_plan() -> Result<()> {
 }
 
 fn make_topk_context() -> ExecutionContext {
-    let config = ExecutionConfig::new().with_query_planner(Arc::new(TopKQueryPlanner {}));
+    let config = ExecutionConfig::new()
```

Comment:
For other reviewers, here is a good example showing how the interface changes. Previously using custom optimizers meant you had to effectively implement a custom "query planner".

I like it. 👍